### ISSUE TITLE
nrunner: avocado-runner-exec and avocado-runner-tap are not passing PATH env

### ISSUE THREAD

**Issue by beraldoleal** [avocado-framework/avocado#3667]
Sometimes, third-party applications and/or tests will call binaries without a full path. They need a previous path to be set.

When running an `avocado-runner-exec` or `avocado-runner-tap` we use `kwargs` as environment variables. But `env` argument from `subprocess.run()` will pass to the subprocess only the desired dict. This will make some tests to fail.

Ex:

```perl
#!/usr/bin/perl
# /tmp/test.pl
# This perl script it is just a test to print current ip addresses
system('ip addr show lo')
```

And if we try to run this with `avocado-runner-exec` without any `kwargs` it will work, because `subprocess.run()` will keep the current environment.

```
(avocado-new) $ avocado-runner-exec runnable-run -k exec -u perl -a /tmp/test.pl
{'status': 'running', 'time_start': 1584647634.7147489}
{'status': 'running'}
{'status': 'finished', 'returncode': 0, 'stdout': b'1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000\n    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00\n    inet 127.0.0.1/8 scope host lo\n       valid_lft forever preferred_lft forever\n    inet6 ::1/128 scope host \n       valid_lft forever preferred_lft forever\n', 'stderr': b'', 'time_end': 1584647634.726349}
```

Now, if I try to run passing any `kwarg`, it will not run the `ip addr show lo` command:

```
(avocado-new) $ avocado-runner-exec runnable-run -k exec -u perl -a /tmp/test.pl FOO='bar'
{'status': 'running', 'time_start': 1584647717.5144787}
{'status': 'running'}
{'status': 'finished', 'returncode': 0, 'stdout': b'', 'stderr': b'', 'time_end': 1584647717.5260687}
```

Note, that besides the `returncode` is 0, `stdout` and `stderr` are empty, and in some cases will fail. This is just an example.

Thinking on the usability, we could do this small task to facilitate the users' life, avoiding to pass a PATH every time. But I'm not sure if this is acceptable.

If this is acceptable, there is a PR already open (currently in draft mode): #3666

**Comment by beraldoleal** [comment]
@clebergnu based on your comment here #3666 I can't see right now how we would solve that problem without exporting the path. So IIUC, your idea is to close this and keep as it is?

**Comment by clebergnu** [comment]
@beraldoleal actually, given that the environment variable (PATH) is *not* being set on the task, it seems that it will affect the reproducibility aspects, which I was worried about in #3666. The PATH will be dynamic whenever it's not set, so I take back my vague "NACK".

**Comment by beraldoleal** [comment]
@clebergnu So, IIUC, are you ok with #3666 being merged?

**Comment by clebergnu** [comment]
> @clebergnu So, IIUC, are you ok with #3666 being merged?

Yes, I think so.  Could you please rebase / resend it so it goes over CI again?

**Comment by clebergnu** [comment]
Resolved with #3666
